application/server/configuration.go

```diff
@@ -150,22 +154,63 @@ func updateAuthenticationMethod(settings lsp.Settings) {
 	c := config.CurrentConfig()
 	c.SetAuthenticationMethod(settings.AuthenticationMethod)
 	if config.CurrentConfig().AuthenticationMethod() == lsp.OAuthAuthentication {
-		engine := c.Engine()
-		conf := engine.GetConfiguration()
-		httpClient := c.Engine().GetNetworkAccess().GetUnauthorizedHttpClient()
-		openBrowserFunc := func(url string) {
-			di.AuthenticationService().Provider().SetAuthURL(url)
-			snyk.DefaultOpenBrowserFunc(url)
-		}
-		authenticator := auth.NewOAuth2AuthenticatorWithCustomFuncs(conf, httpClient, openBrowserFunc, auth.ShutdownServer)
-		oAuthProvider := oauth.NewOAuthProvider(conf, authenticator)
-		di.AuthenticationService().SetProvider(oAuthProvider)
+		configureOAuth(c, auth.RefreshToken)
 	} else {
 		cliAuthenticationProvider := auth2.NewCliAuthenticationProvider(di.ErrorReporter())
 		di.AuthenticationService().SetProvider(cliAuthenticationProvider)
 	}
 }
 
+func credentialsUpdateCallback(_ string, value any) {
+	newToken, ok := value.(string)
+	if !ok {
+		msg := fmt.Sprintf("Failed to cast token value of type %T to string", value)
+		log.Error().Str("method", "storage callback token").
+			Msgf(msg)
+		di.ErrorReporter().CaptureError(errors.New(msg))
+		return
+	}
+	bytes, err := json.Marshal(newToken)
```

Comment:
issue: newToken is already a string and can be directly used in the calls below, json.Marshal() is not necessary.